plugins/inputs/logfile/logfile_test.go

```diff
@@ -738,42 +742,44 @@ func TestLogsFileRecreate(t *testing.T) {
 		t.Fatalf("%v log src was returned when 1 should be available", len(lsrcs))
 	}
 
+
 	lsrc := lsrcs[0]
+	defer lsrc.Stop()
 	evts := make(chan logs.LogEvent)
 	lsrc.SetOutput(func(e logs.LogEvent) {
 		if e != nil {
 			evts <- e
 		}
 	})
 
-	go func() {
```

Comment:
Previously, it looks like they were performing the file delete and recreate in a thread, but first waiting 1 second do the activity.  This allowed for the `for` loop below to kick off `tt.FindLogSrc()` at least 1 iteration before `err = os.Remove(tmpfile.Name())` was run.

Reply:
I don't think we need this loop to run before the file is deleted.
```
for {
		lsrcs = tt.FindLogSrc()
		if len(lsrcs) > 0 {
			break
		}
		time.Sleep(1 * time.Second)
	}
```

The loop is just waiting for a new LogSrc to be detected.
This should happen after the file has been removed and then recreated.
I saw no reason not to this in a single-threaded-manner.
I added comments and log messages to explain what the test is doing.
Please let me know if you think those are inaccurate, or just too noisy and should be removed.